From ea9d57a00e4429935dd224bcc4267d501cb43594 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Thu, 8 Aug 2019 23:46:49 +0300
Subject: [PATCH] Fix import order

SageMath is usually unhappy if we do not import sage.all first
---
 flatsurf/geometry/polygon.py | 14 +-------------
 1 file changed, 1 insertion(+), 13 deletions(-)

diff --git a/flatsurf/geometry/polygon.py b/flatsurf/geometry/polygon.py
index 829c511..bef0ed7 100644
--- a/flatsurf/geometry/polygon.py
+++ b/flatsurf/geometry/polygon.py
@@ -26,23 +26,11 @@ from six.moves import range, map, filter, zip
 
 import operator
 
-from sage.misc.cachefunc import cached_method
+from sage.all import cached_method, Parent, UniqueRepresentation, Sets, Fields, ZZ, QQ, AA, RR, matrix, polygen, vector, free_module_element
 
 from sage.structure.element import Element
-from sage.structure.parent import Parent
-from sage.structure.unique_representation import UniqueRepresentation
-
-from sage.categories.sets_cat import Sets
-from sage.categories.fields import Fields
 from sage.categories.action import Action
 
-from sage.rings.all import ZZ, QQ, AA, RR
-
-from sage.modules.free_module_element import vector
-from sage.modules.free_module_element import free_module_element
-from sage.matrix.constructor import matrix
-from sage.rings.polynomial.polynomial_ring import polygen
-
 from .matrix_2x2 import angle
 
 # we implement action of GL(2,K) on polygons
-- 
2.22.0

